### PR TITLE
[Feature] Adding human render to gym wrapper

### PR DESCRIPTION
**Pull request by vincentpierre**
## Motivation and Context

 - Removing `HabRenderWrapper` since it seems it only adds extra metrics to the rendering (these metrics should be added to the task configuration if they are needed). Please advise
 - Using pygame to render the environment in human mode in a similar manner to interactive play. Made it so that if pygame is not installed, human render mode will not be usable. Please advise.
 - Future? Remove the HabitatRenderXXX default gym tasks. These add the third RGB camera as an observation so I find adding Render to the name is misleading. Thoughts?

## Types of changes

- Docs change / refactoring / dependency upgrade
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by ASzot — COMMENTED** [review]
Looks good, but we should have test for human render mode.

**Comment by vincentpierre** [comment]
> Looks good, but we should have test for human render mode.

YES x 1000.
I think I would like some advice on that front. I could add pygame to CI to try to make a test there BUT it seems hard to make sure rendering works properly. At best we can test for errors but it will be hard to test the rendering is actually working properly...

**Comment by ASzot** [comment]
I think a good first step is adding `pygame` to CI and then just checking for errors.

**Comment by vincentpierre** [comment]
Ok, I added a test. Looks like pygame is already on CI.

File: test/test_gym_wrapper.py [hunk 1]
```diff
@@ -183,7 +187,8 @@ def test_auto_gym_wrapper(test_cfg_path):
     config = habitat.get_config(test_cfg_path)
     if "GYM" not in config or config.GYM.AUTO_NAME == "":
         pytest.skip(f"Gym environment name isn't set for {test_cfg_path}.")
-
+    if not pygame_installed:
```
Comment (Skylion007) [inline test/test_gym_wrapper.py:190]: use pytest importorskip https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-importorskip

**Comment by vincentpierre** [comment]
I do not think pygame cannot run on CI because there is no monitor. This will make this feature very hard to test on CI. Do you have suggestions?

**Comment by vincentpierre** [comment]
I will be using Mock to replace pygame in the test. This way I will avoid having to open the pygame window on the CI machine. It will also prevent having the pygame window open when testing locally, which would shift the cursor focus (would be unnecessarily distracting)

File: habitat/utils/gym_adapter.py [hunk 2]
```diff
@@ -294,17 +301,37 @@ def reset(self) -> Union[np.ndarray, Dict[str, np.ndarray]]:
...
+                )
+            frame = observations_to_image(
+                self._last_obs, self._env._env.get_metrics()
+            )
+            if self._screen is None:
+                pygame.init()
+                self._screen = pygame.display.set_mode(
+                    [frame.shape[1], frame.shape[0]]
+                )
+            draw_frame = np.transpose(frame, (1, 0, 2))
```
Comment (mathfac) [inline habitat/utils/gym_adapter.py:323]: That's great to add a comment like  `(C, H,  W) -> (H, C, W)` for transpose operations.

File: habitat/utils/gym_adapter.py [hunk 3]
```diff
@@ -294,17 +301,37 @@ def reset(self) -> Union[np.ndarray, Dict[str, np.ndarray]]:
...
+            draw_frame = pygame.surfarray.make_surface(draw_frame)
+            self._screen.fill((0, 0, 0))  # type: ignore[attr-defined]
```
Comment (mathfac) [inline habitat/utils/gym_adapter.py:325]: Let's avoid magic unnamed numerical consts:
```suggestion
            BLACK_COLOR = (0, 0, 0)
            self._screen.fill(BLACK_COLOR)  # type: ignore[attr-defined]
```

**Review by mathfac — APPROVED** [review]
Overall looks good, some light comments.

**Pull request merged**